CodeU_Assignement_5/src/codeu_assignement_5/KemUnknownLanguage.java

```diff
@@ -0,0 +1,145 @@
+package codeu_assignement_5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Karine
+ */
+public class KemUnknownLanguage {
+    /*
+     * Input: a dictionary (a list of words in lexicographic order) of all words in an unknown/invented language, 
+    *  Output: alphabet (an ordered list of characters) of that language, letters we cannot decide are appended in the end.
+    */
+    public List<Character> getAlphabeit(String[] alienWords) {
+        if ((alienWords != null) && (alienWords.length > 0)) {
+           return extractAlphabeitFromWords(alienWords);
+        }
+        
+        return null;
+    }
+
+    private List<Character> extractAlphabeitFromWords(String[] alienWords) {
+        /* Get the current order by prefix size, starting from 0 */
+        List<Character> dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+        /* A,R,C from the example! */
+
+        /* Add from prefix while there are letters after first letter of prefix */
+        while (!addToDictByPrefix(alienWords, dict)){} 
+
+        /* Add all the chars we cannot decide in the end*/
+        for (String w : alienWords) {
+            for (Character c: w.toCharArray()) {
+                if (!dict.contains(c)) {
+                    dict.add(c);
+                }
+            }
+        }
+        
+        return dict;
+    }
```

Comment:
I'm afraid I just don't understand the algorithm you're attempting here, so I'm stopping the review at this point.

Reading code shouldn't be hard work, even if it's harder to write to make it easy to read.

I have a strong suspicion that the code will have cases that don't work - what I can make out doesn't suggest an algorithm that I think will work in all corner cases (having done a couple of code reviews with nasty corner case failures for other mentees already).

What I suggest is to extend the commenting on the code to express what you're expecting at each stage of the algorithm, any invariants you might be wanting to maintain, etc. If you can break your algorithm into a clear pipeline of steps, that will also help. Take a look at the code written by the others to get ideas on how to cleanly express your intent.